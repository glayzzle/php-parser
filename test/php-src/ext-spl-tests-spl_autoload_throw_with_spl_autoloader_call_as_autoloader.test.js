// eslint-disable prettier/prettier
const parser = require("../main");

describe("php-src tests", function () {
  // ext/spl/tests/spl_autoload_throw_with_spl_autoloader_call_as_autoloader.phpt
  it("spl_autoload_register() function - warn when using spl_autoload_call() as the autoloading function", function () {
    expect(parser.parseCode("<?php\ntry {\n    spl_autoload_register('spl_autoload_call');\n} catch (\\ValueError $e) {\n    echo $e->getMessage() . \\PHP_EOL;\n}\n?>")).toMatchSnapshot();
  });
});
