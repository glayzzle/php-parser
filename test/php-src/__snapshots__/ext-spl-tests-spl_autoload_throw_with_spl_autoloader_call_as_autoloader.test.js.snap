// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`php-src tests spl_autoload_register() function - warn when using spl_autoload_call() as the autoloading function 1`] = `
Program {
  "children": Array [
    Try {
      "always": null,
      "body": Block {
        "children": Array [
          ExpressionStatement {
            "expression": Call {
              "arguments": Array [
                String {
                  "isDoubleQuote": false,
                  "kind": "string",
                  "raw": "'spl_autoload_call'",
                  "unicode": false,
                  "value": "spl_autoload_call",
                },
              ],
              "kind": "call",
              "what": Name {
                "kind": "name",
                "name": "spl_autoload_register",
                "resolution": "uqn",
              },
            },
            "kind": "expressionstatement",
          },
        ],
        "kind": "block",
      },
      "catches": Array [
        Catch {
          "body": Block {
            "children": Array [
              Echo {
                "expressions": Array [
                  Bin {
                    "kind": "bin",
                    "left": Call {
                      "arguments": Array [],
                      "kind": "call",
                      "what": PropertyLookup {
                        "kind": "propertylookup",
                        "offset": Identifier {
                          "kind": "identifier",
                          "name": "getMessage",
                        },
                        "what": Variable {
                          "curly": false,
                          "kind": "variable",
                          "name": "e",
                        },
                      },
                    },
                    "right": Name {
                      "kind": "name",
                      "name": "\\\\PHP_EOL",
                      "resolution": "fqn",
                    },
                    "type": ".",
                  },
                ],
                "kind": "echo",
                "shortForm": false,
              },
            ],
            "kind": "block",
          },
          "kind": "catch",
          "variable": Variable {
            "curly": false,
            "kind": "variable",
            "name": "e",
          },
          "what": Array [
            Name {
              "kind": "name",
              "name": "\\\\ValueError",
              "resolution": "fqn",
            },
          ],
        },
      ],
      "kind": "try",
    },
  ],
  "errors": Array [],
  "kind": "program",
}
`;
